src/main/java/server/Server.java

```diff
@@ -0,0 +1,102 @@
+package server;/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+
+import Response.Response;
+
+import java.io.*;
+import java.net.*;
+import java.util.ArrayList;
+
+import static com.google.common.io.ByteStreams.toByteArray;
+
+public class Server {
+  private PrintWriter out;
+  private BufferedReader in;
+  private ServerSocket serverSocket;
+  private int PORT = 3000;
+  private Socket clientSocket;
+
+  public void go() throws IOException {
+    serverSocket = createSocket();
+
+    while (true) {
+      clientSocket = clientSocket(serverSocket);
+
+      out = createWriter(clientSocket);
+      in = createReader(clientSocket);
+      handleRequest(in, out);
+      closeConnection(in, out);
```

Comment:
I see in other parts of this file that you're invoking _both_ input and output in `handleRequest` and `closeConnection`.  I've made some suggestions to decouple those a little, but just to illustrate what I'm thinking of, the shape might be something like this:

```java
Socket clientSocket = serverSocket.accept();
BufferedReader input = createReader(clientSocket);
PrintWriter output = createWriter(clientSocket);

// This would turn the request into a `RequestContent` Java object.
//
RequestContent request = readRequest(input);

// This accepts a RequestContent object, and creates a custom ResponseContent object
//
ResponseContent response = new ResponseContent(request);

// This writes a response to the output
//
output.write(response.toString());

clientSocket.close();
```